### ISSUE TITLE
booting me out when trying to run backup

### ISSUE THREAD

**Issue by trilanders** [rmcrackan/Libation#224]
everything was working up till last night. was running 6.7.4 tried rolling back to 6.7.2 and got the same issue. turned on extended logging and looked through the logs. last log entry was:
 {}2022-04-07 06:06:16.188 -04:00 [INF] (at Dinah.Core.StepRunner.BaseStep.logBegin()) Begin step 'Step 2: Download Decrypted Audiobook'
i don't know if there is any private information in the logs so i didn't want to upload the full log before i hear from you.
basically what is happening is i start the backup and it boots me out of the program.

any ideas? or is there a private email i can send the full log to or is it safe to put it here?

**Comment by rmcrackan** [comment]
I've tried hard to keep the logs free of personal information. They're safe to attach here.

**Comment by trilanders** [comment]
ok here is a small one. i reset the logs to spit out everything, opened it up and clicked on a single book at a time until it booted me out of the program.
[Log202204.log.txt](https://github.com/rmcrackan/Libation/files/8444561/Log202204.log.txt)

**Comment by wtanksleyjr** [comment]
This looks like what I've been experiencing too -- Audible seems to be
noticing huge downloads and expiring authentication. Perhaps a partial fix
might be to detect that it's an authentication issue and re-authenticate,
instead of failing all of the books in a row.

From the user side, when you see this, does it work to re-scan your library
(it'll ask you to reenter your password) and then restart liberation?
That's what I did.

-Wm

**Comment by trilanders** [comment]
with the new update before doing a scan is 1st.log and after doing a scan is 2nd.log. still booting me out and i noticed that the file it fails on is left in the temp folder.
[1st.log](https://github.com/rmcrackan/Libation/files/8454585/1st.log)
[2nd.log](https://github.com/rmcrackan/Libation/files/8454586/2nd.log)

**Comment by rmcrackan** [comment]
The errors I see in the logs are for books that you don't have permissions for. Were these returned? Or were they audible plus titles that you removed from you library? To quickly find and remove books that are no longer in your library, you can go to Import > Remove Library Books

**Comment by trilanders** [comment]
See that is the strange thing… all of them are in my library. If I go into the mobile app I am able to download them. If I scan for new books and try to download one of the new books that I pick it does the same thing.

**Comment by rmcrackan** [comment]
Ah, now I understand. What if you go to accounts, delete your account, and save. Go back into accounts and re-add, save, then try again.

**Comment by rmcrackan** [comment]
Any update on this issue?

**Comment by trilanders** [comment]
Sorry haven’t had the chance to troubleshoot more. Just got out of the hospital earlier today. At last try i found out that if i was playing an audio book on either my phone or alexa device it would run a full backup with no errors. Once i stopped playing the book if i tried to do the backup i got the same error about not having access to that book.

I’ll do some more testing over the weekend, but it looks like amazon is expiring my key unless it is currently active.

**Comment by rmcrackan** [comment]
Wow, that's a new one. Let me know if you can find anything else. I'm not even sure how to start trouble shooting this on my end.

I hope getting out of the hospital is a good thing. Best wishes.

**Comment by rmcrackan** [comment]
I've been unable to replicate this on my side. Any progress on your end?

**Comment by trilanders** [comment]
I’ll give that a try later tonight, but I figured that I would update you. I don’t know if this has anything to do with it, but while listening on an Alexa device and without doing a library scan I was able to do a full backup with no errors. I gave it a couple of hours after doing the last scan and getting an error and then tried it this way. I am wondering if the Alexa device is somehow keeping the session alive…
